libcxx/test/std/containers/sequences/array/array.overview/nttp.verify.cpp

```diff
@@ -0,0 +1,80 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// UNSUPPORTED: c++03, c++11, c++14, c++17
+
+// <array>
+
+// array<T, N> is a structural type ([temp.param]) if T is a structural type.
```

Comment:
I don't understand what you're testing here. Isn't it sufficient to check for the NTTP behavior as far as the LWG issue is concerned? I think I'm just missing the obvious here, but if you can explain that would be great!

Reply:
I'm testing whether `array` (or `pair`) is a structural type based on its template parameters. LWG-3382 states that:
* pair<T, U> is a structural type (13.2 [[temp.param]](https://wg21.link/temp.param)) if T and U are both structural types
* array<T, N> is a structural type (13.2 [[temp.param]](https://wg21.link/temp.param)) if T is a structural type

IMHO those tests address the LWG issue, although they might be an overkill.